src/GovUk.Education.ExploreEducationStatistics.Data.Services/SubjectMetaService.cs

```diff
@@ -102,30 +102,44 @@ private SubjectMetaViewModel GetSubjectMetaViewModel(Subject subject)
         private SubjectMetaViewModel GetSubjectMetaViewModelFromQuery(SubjectMetaQueryContext query)
         {
             var observations = _observationService.FindObservations(query).AsQueryable();
-
+            var locations = new Dictionary<string, ObservationalUnitsMetaViewModel>();
+            var timePeriods = new TimePeriodsMetaViewModel();
+            var filters = new Dictionary<string, FilterMetaViewModel>();
+            var indicators = new Dictionary<string, IndicatorsMetaViewModel>();
+            
             var stopwatch = Stopwatch.StartNew();
             stopwatch.Start();
 
-            var filters = query.TimePeriod != null ? GetFilters(query.SubjectId, observations, false) : new Dictionary<string, FilterMetaViewModel>();
-
-            _logger.LogTrace("Got Filters in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
-            stopwatch.Restart();
-
-            var indicators = query.TimePeriod != null ? GetIndicators(query.SubjectId) : new Dictionary<string, IndicatorsMetaViewModel>();
+            if (query.Locations == null)
+            {
+                locations = GetObservationalUnits(observations);
+                
+                _logger.LogTrace("Got Observational Units in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+                stopwatch.Restart();
+            }
+            
+            if (query.TimePeriod == null && query.Locations != null)
+            {
+                timePeriods = query.TimePeriod != null ? null : GetTimePeriods(observations);
 
-            _logger.LogTrace("Got Indicators in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
-            stopwatch.Restart();
+                _logger.LogTrace("Got Time Periods in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+                stopwatch.Restart();
+            }
+            
+            if (query.TimePeriod != null)
+            {
+                filters = query.TimePeriod != null ? GetFilters(query.SubjectId, observations, false) : new Dictionary<string, FilterMetaViewModel>();
```

Comment:
`query.TimePeriod != null` is always true here

Reply:
aah yes - thanks - done